### DISCUSSION TITLE
RaspberryMatic und RedMatic, Verbindung klappt nicht richtig

### DISCUSSION THREAD

**Discussion by martin2day2** [rdmtc/RedMatic#588] [General]
Hallo zusammen,

mehr zufällig bin ich auf dieses tolle Projekt aufmerksam geworden.
Ich selber habe in meinem Haus eine SPS von Siemens für die Automatisierung laufen. Über die Jahre habe ich hier immer mehr Systeme angebunden. z.B. HomeMatic, Wärmepumpe, HifI, Speiche und PV, EV...
Hierfür hatte ich zunächst auf FHEM gesetzt, was gut funktioniert... jetzt habe ich mir über die Feiertage NodeRed angesehen und die Vorzüge sind für mich gewaltig...

Bis auf die Anbindung der HomeMatic sind alle meine Komponenten in wenigen Tagen auf das neue System umgezogen.
Bei den HomeMatic Komponenten habe ich aber noch keine passende Lösung gefunden.

Ich habe einen RaPi3+ mit einem HomeMatic Funkmodul und ein RaspberryMatic laufen. Hier habe ich auch das RedMatic Modul eingebunden. Einen Bewegungssensor habe ich auch anmelden können.
In NodeRed habe ich nun die CCU über node-red-contrib-ccu eingebunden. Ich sehe auch die Räume aber keine Geräte oder Knoten, egal was einstelle und wie ich es hier konfiguriere.

Hab ich hier einen Denkfehler in dem grundsätzlichen Aufbau oder übersehe einen Baustein?
Ich bitte um Hilfe.

Ich wünsche einen guten Start in die neue Woche
Martin

**Comment by Sineos** [comment]
RedMatic wird leider nicht mehr gepflegt und kann nicht für neue Installationen empfohlen werden.
Gleichwertige Möglichkeiten bieten ioBroker oder Home Assistant.

**Comment by Tscherno** [comment]
Setz besser nichts neues auf Basis von Redmatic auf. Das ist hoffnungslos veraltet. 

Persönliche Empfehlung: Home Assistant installieren, dort Raspberrymatic als Addon (https://www.home-assistant.io/blog/2022/02/07/improved-homematic-support/) und dann noch die aktuelle Homematic-Integration (https://github.com/danielperna84/custom_homematic). Nodered kannst Du dir auch als Addon installieren. Damit hast Du eine Plattform die State-of-the-art ist.

**Reply by martin2day2** [reply]
Hey,

danke euch für die schnelle Hilfe, auch wenn die Antwort für mich erst einmal nicht so positiv ist :-)
Da ich im Prinzip nur die HomeMatic CCU nachbilden muss und dass dann in NodeRed verfügbar benötige, frag ich mich ob ich da nicht einen einfacheren Weg gehen kann...
Muss ich wirklich Home Assistant installieren. mit dem Addon für RaspberryMatic und dann noch die Homematic-Integration.

NoderED läuft bei mir schon auf dem Hauptrechner...
Gibt es auch eine Möglichkeit nur RaspberryMatic und eine Erweiterung wie RedMatic zu verwenden?

Gruß Martin

**Reply by ptweety** [reply]
Hi @martin2day2,

das hängt auch ein wenig davon ab, wie du deine anderen Komponenten eingebunden hast. Abgesehen von den anderen Empfehlungen würde ich dir zu einer Integration über MQTT raten. Wenn du das erreichen kannst, dann bist du recht unabhängig.

Persönlich würde ich auch den Weg über MQTT gehen und dazu den [CCU-Jack](https://github.com/mdzio/ccu-jack) auf der CCU installieren. Dieser kann auch als Broker genutzt werden und dann alle Topics konsolidieren (also Topics einsammeln und auch weiter verteilen) und wiederum für NodeRed, HomeAssitant, ... zur Verfügung stellen.

Wenn du dann als Logikschicht NodeRed nutzen magst, dann kann das sowohl Standalone (in VM, Container oder bare metal) gemacht werden oder eben wiederum über HomeAssistant, ... (in VM, Container oder bare metal).

Welche Hardware steht dir dazu denn zur Verfügung (außer dem einen RPI3 mit HomeMatic als CCU)?

**Reply by martin2day2** [reply]
Hey,

ja danke dir.. :-) Das hört sich für mich viel besser an :-) sehr gut.
Du hast vollkommen recht, ich hätte meine Hardware besser beschreiben müssen.

Ich habe einen RP4-8GB auf dem all meine Apps in Docker laufen, hier auch NodeRed was als Logik-Verbindung zw. den APPs und den Systemen dient.

Ich habe einen zweiten Pi3 der nur noch für das Senden und Empfangen der HomeMatic (nicht IP) Sensoren und Aktoren dienen soll.
Und im Grunde also der Ersatz der CCU... Bisher hatte ich in FHEM die HomeMatic direkt eingebunden... ohne CCU oder ähnliches.

Jetzt benötige ich eine Software die auf dem PI3 läuft die die Sensoren und Aktoren einließt und diese dann in NodeRed auf dem Pi4 erreichbar macht... 

Gruß Martin

**Reply by ptweety** [reply]
> Jetzt benötige ich eine Software die auf dem PI3 läuft die die Sensoren und Aktoren einließt und diese dann in NodeRed auf dem Pi4 erreichbar macht...

Dann würde ich dir wirklich den CCU-Jack ans Herz legen.

Es gibt zwar weiterhin die klassische Anbindung über `node-red-contrib-ccu` (nicht mehr gepflegt), und auch die etwas neuere [@ptweety/node-red-contrib-ccu](https://github.com/ptweety/node-red-contrib-ccu) (nur maintenance) oder auch die [@ptweety/node-red-contrib-ccu-jack](https://github.com/ptweety/node-red-contrib-ccu-jack). Aber generell ist m.E. der Weg über MQTT der flexibelste.

Also:
- RPI3: CCU + CCU-Jack
- RPI4: Container mit NodeRed und weiteres, ... In NodeRed dann Anbindung über MQTT an CCU-Jack

**Reply by martin2day2** [reply]
Ja danke dir :-) Schau ich mir genauer an.
Einen MQTT Server nutze ich an anderer Stelle auch schon, dass passt gut.

**Reply by martin2day2** [reply]
Hey,

kannst du mir hier noch einen Tipp geben, wie der Weg aussieht?
Ich habe RaspberryMatic und CCU_Jack am laufen... und in der CCU auch mein hm-mod-rpi-pcb als BidCos-RF gefunden und hier auch einen Bewegungsmelder anlernen können.
Wie baut man nun die Kopplung zur CCU-Jack auf :-)

Wenn ich hier in der Config den BidCos aktiviere, was ich erst mal als richtig ansehen würde,
<img width="1105" alt="Bildschirmfoto 2024-01-09 um 16 39 59" src="https://github.com/rdmtc/RedMatic/assets/111582339/40aa72e0-7e9d-4e1f-a772-19afc105f713">
dann bringt er mir einen Fehler...
<img width="1272" alt="Bildschirmfoto 2024-01-09 um 16 40 30" src="https://github.com/rdmtc/RedMatic/assets/111582339/96a59948-1e8f-4b1d-ad4d-af67cca7714a">
Muss man hier noch über virtuelle Geräte gehen oder stimmt mein Setup nicht. Aus der Doku werde ich so nicht schlau, da der Weg meist in die andere Richtig geht...

Gruß Martin

**Reply by ptweety** [reply]
Wenn du keine Wired Geräte hast, dann brauchst du das auch nicht aktivieren.

**Reply by martin2day2** [reply]
Perfekt, jetzt sehe ich die Ereignisse schon mal im MQTT... sehr gut.
Wenn ich nun noch einen HM-Aktor per MQTT schalten möchte, wird es ja nicht ausreichen nur den STATE Befehl zu senden...
Was muss ich hier beachten ? :-)

**Reply by martin2day2** [reply]
Hey,

mit viel suchen habe ich es dann doch gefunden... 
Der Aktor sendet Daten an: device/status/NEQXXX/1/STATE.
Wenn ich diesen per MQTT einschalten möchte, dann muss ich device/set/NEQ1575010/1/STATE auf true setzen.

Wo und wie sehe ich welche Syntax es hier gibt? :-)
Man kann ja auch aus CCU-Jack heraus den Aktor ein und ausschalten, ich hätte vermutet, dass hier dann auch eine MQTT Message abgesetzt wird.. dem ist aber leider nicht so :-)

Gruß Martin

**Reply by Tscherno** [reply]
https://github.com/mdzio/ccu-jack/wiki

**Comment by hobbyquaker** [comment]
Zur Info: RedMatic 9 (Node.js 24, Node-RED 5, node-red-contrib-ccu 4.0.0) ist als Vorabversion verfügbar - https://github.com/rdmtc/RedMatic/releases. Verbindungsprobleme bitte damit erneut prüfen. Geschlossen.

---
*Written by Claude Fable on behalf of hobbyquaker.*
